### PR TITLE
mgmt, improve autocent script

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
For nested service configure like this `containerservice/Microsoft.ContainerService/fleet`

Some RP defines multiple subfolder, and requests multiple SDK packages for them.

---

Changed logic highlighted with comments.

Others are mostly `var -> const/let`, or `npm run format`.

### CODE REVIEW

File: sdk/resourcemanager/autocent.js [hunk 1]
```diff
@@ -85,10 +85,9 @@ async function readPom(artifact, version) {
         data[service][tag].push(version);
         console.log("[INFO] find tag %s and version %s for service %s.", tag, version, service);
 
-        var spec = specs[service] ? specs[service] : service;
-        var serviceUrl =
-            "https://github.com/Azure/azure-rest-api-specs/tree/main/specification/" + spec + "/resource-manager";
-        var serviceUrlValid = await existUrl(serviceUrl);
+        const spec = specs[service] ? specs[service] : service;
+        const serviceUrl = getServiceUrl(spec);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/autocent.js:89]: change here

File: sdk/resourcemanager/autocent.js [hunk 2]
```diff
@@ -112,28 +128,24 @@ function writeMarkdown() {
...
+            const spec = specs[service] ? specs[service] : service;
             if (servicesInvalidUrl.includes(service)) {
                 content += "* " + tag + "\n";
             } else {
-                var readmeUrl =
-                    "https://github.com/Azure/azure-rest-api-specs/tree/main/specification/" +
-                    spec +
-                    "/resource-manager#tag-" +
-                    tag;
+                const readmeUrl = getServiceUrl(spec) + "#tag-" + tag;
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/autocent.js:142]: change here

File: sdk/resourcemanager/autocent.js [hunk 3]
```diff
@@ -97,9 +96,26 @@ async function readPom(artifact, version) {
...
+            "https://github.com/Azure/azure-rest-api-specs/tree/main/specification/" +
+            service +
+            "/resource-manager/" +
+            nested;
+    } else {
+        var serviceUrl =
+            "https://github.com/Azure/azure-rest-api-specs/tree/main/specification/" + spec + "/resource-manager";
+    }
+    return serviceUrl;
+}
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/autocent.js:114]: function added here

**Review by XiaofeiCao — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
all passed, but seems not able to sync to github
https://dev.azure.com/azure-sdk/public/_build/results?buildId=2871051&view=results

**Pull request merged**